File: Fase 2/revistas-app/src/app/objetos/base/Usuario.ts
import { Etiqueta } from "./Etiqueta";
import { GeneroEnum } from "./GeneroEnum";

export class Usuario {
    private _username: string;
    private _password: string;
    private _descripcion: string | null;
    private _hobbies: string | null;
    private _genero: GeneroEnum | null;
    private _etiquetas: Array<Etiqueta>;
    
    constructor(username: string, password: string, descripcion: string, hobbies: string, genero: GeneroEnum, etiquetas: Array<Etiqueta>){
        this._username = username;
        this._password = password;
        this._descripcion = descripcion;
        this._hobbies = hobbies;
        this._genero = genero;
        this._etiquetas = etiquetas;
    }
    
    public get username(): string {
        return this._username;
    }
    public set username(value: string) {
        this._username = value;
    }

    public get password(): string {
        return this._password;
    }
    public set password(value: string) {
        this._password = value;
    }

    public get descripcion(): string|null {
        return this._descripcion;
    }
    public set descripcion(value: string|null) {
        this._descripcion = value;
    }

    public get hobbies(): string|null {
        return this._hobbies;
    }
    public set hobbies(value: string|null) {
        this._hobbies = value;
    }

    public get genero(): GeneroEnum|null {
        return this._genero;
    }
    public set genero(value: GeneroEnum|null) {
        this._genero = value;
    }

    public get etiquetas(): Array<Etiqueta> {
        return this._etiquetas;
    }
    public set etiquetas(value: Array<Etiqueta>) {
        this._etiquetas = value;
    }
}

File: Fase 2/ServidorRevistas/src/main/java/objetos/Usuario.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package objetos;

import java.util.ArrayList;

/**
 *
 * @author fernanrod
 */
public class Usuario {
    private String username;
    private String password;
    private String descripcion;
    private String hobbies;
    private GeneroEnum genero;
    private ArrayList<Etiqueta> etiquetas;
    
    public Usuario(String username,String password,String descripcion,String hobbies,GeneroEnum genero,ArrayList<Etiqueta> etiquetas){
        this.username = username;
        this.password = password;
        this.descripcion = descripcion;
        this.hobbies = hobbies;
        this.genero = genero;
        this.etiquetas = etiquetas;
    }
    
    public String getUsername() {
        return username;
    }

    public String getPassword() {
        return password;
    }

    public String getDescripcion() {
        return descripcion;
    }

    public String getHobbies() {
        return hobbies;
    }

    public GeneroEnum getGenero() {
        return genero;
    }

    public ArrayList<Etiqueta> getEtiquetas() {
        return etiquetas;
    }
}
